core/src/testils/mod.rs

```diff
@@ -13,16 +13,36 @@ pub mod byte_arrays {
 
 pub use self::packet::*;
 pub use self::rvg::*;
-pub use crate::dpdk::{Mempool, SocketId, MEMPOOL};
 
-use crate::dpdk::eal_init;
+use crate::dpdk::{self, Mempool, SocketId, MEMPOOL};
+use std::ptr;
 use std::sync::Once;
 
 static TEST_INIT: Once = Once::new();
 
-/// Run once initialization of EAL for `cargo test`
+/// Run once initialization of EAL for `cargo test`.
 pub fn cargo_test_init() {
     TEST_INIT.call_once(|| {
-        eal_init(vec!["nb2_test".to_owned()]).unwrap();
+        dpdk::eal_init(vec!["nb2_test".to_owned()]).unwrap();
     });
 }
+
+/// A handle that keeps the mempool in scope for the duration of the test. It
+/// will unset the thread bound mempool on drop.
+#[allow(dead_code)]
+pub struct MempoolGuard {
+    inner: Mempool,
+}
+
+impl Drop for MempoolGuard {
+    fn drop(&mut self) {
+        MEMPOOL.with(|tls| tls.replace(ptr::null_mut()));
+    }
+}
+
+/// Creates a new mempool for test.
+pub fn new_test_mempool(capacity: usize, cache_size: usize) -> MempoolGuard {
```

Comment:
minor: I'm ok w/ this, but if we stick w/ this approach, I'd say name it `new_mempool` as it is exported through testils already. Otherwise, would it be cleaner to have a compiled-under-test `new` static function under mempool (w/ the guard) that gets used, and then it's just the ame new but for tests? I'm not a stickler for it though and know how much you love feature flags :).